Generate Software Bill of Materials during release (#22191)
cilium-team/cilium: Documentation/configuration/sbom.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _sbom:

**************************
Software Bill of Materials
**************************

Since version 1.13, all Cilium images include a Software Bill of Materials (SBOM). 

> We use the `bom`_ tool to generate the SBOM.

Reviewer: nit: the guide is not a person, ideally it's written in a factual manner:
```suggestion
This SBOM is generated by the `bom`_ tool.
```

.. _`bom`: https://github.com/kubernetes-sigs/bom 

Prerequisites
=============

- `Install cosign`_

.. _`Install cosign`: https://docs.sigstore.dev/cosign/installation/

Download SBOM
=============

The SBOM can be downloaded from the supplied Cilium image using the ``cosign download sbom`` command.

.. code-block:: shell-session

>     $ cosign download sbom --output-file sbom.spdx <IMAGE URL>

Reviewer: Should there be another subsequent step after this? I think that at the moment this documentation page first assumes that the user knows exactly what the SBOM is and what they want to achieve with it. But the steps only say to download it.

I don't think that we need to give a full step through of how to interact with the SBOM - that is presumably already covered somewhere else in sigstore. But it might be useful to add a section for "next steps" - whether it be verifying signatures, evaluating dependencies in the sbom, etc. Feel free to just link to other resources here that give a good explanation, you don't need to rewrite the wheel.

Author: @joestringer Sigstore has instructions for attaching, signing, and downloading an SBOM from an image. I've found many posts about what an SBOM is and why we need one, but not much on how to process an SBOM after downloading it from an image. We can do something like: "To learn more about SBOM see [What an SBOM Can Do for You](https://www.chainguard.dev/unchained/what-an-sbom-can-do-for-you)" at the beginning of the docs on SBOM. Would that be ok?

Reviewer: Sure, I think that makes sense.
